### PR TITLE
Add validations.md

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Adds a file describing the validations required for upgrade 4.

This PR is in draft, as instructions for using DiffyScan are still needed to validate the implementation code.

### CODE REVIEW

File: tasks/eth/2-add-superchainConfig/Validation.md
Comment (mds1) [inline tasks/eth/2-add-superchainConfig/Validation.md:1]: The PR calls this upgrade 4 but this file is called `2-*`, which feels confusing. Should those match at all? Or are the 0/1/2 number prefixes for existing task folder names just communicating the order they were added to the repo for some reason?

File: tasks/eth/2-add-superchainConfig/Validation.md [hunk 1]
```diff
@@ -0,0 +1,146 @@
+# Validation
+
+This document can be used to validate the state diff resulting from the execution of the upgrade
+transaction.
```
Comment (mds1) [inline tasks/eth/2-add-superchainConfig/Validation.md:4]: Can we link to the corresponding governance post?

File: tasks/eth/2-add-superchainConfig/Validation.md [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+
+Please ensure that the following changes (and none others) are made to each contract in the system.
+Validation of the bytecode deployed at the implementation addresses is based on the presence of
+those addresses at the tagged commit in the Optimism monorepo as linked below.
+
+## Pre-upgrade deployed code
+
+The `SuperchainConfigProxy` has already been deployed and initialized. It's address
+`0x95703e0982140d16f8eba6d158fccede42f04a4c` appears in the state diff below in the contracts which
```
Comment (mds1) [inline tasks/eth/2-add-superchainConfig/Validation.md:13]: nit, checksum
```suggestion
`0x95703e0982140D16f8ebA6d158FccEde42f04a4C` appears in the state diff below in the contracts which
```

File: tasks/eth/2-add-superchainConfig/Validation.md [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+  **After:** `0x000000000000000000000000566511a1a09561e2896f8c0fd77e8544e59bfdb0` <br/>
+  **Meaning:** Implementation address is set to the new [`L1StandardBridge`](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0-rc.1/op-chain-ops/cmd/op-upgrade-extended-pause/main.go#L51-L53).
+
+### `0x9ba6e03d8b90de867373db8cf1a58d2f7f006b3a` (`SafeProxy`)
+
+- **Key:** `0x0000000000000000000000000000000000000000000000000000000000000005` <br/>
+  **After:** `0x0000000000000000000000000000000000000000000000000000000000000057` <br/>
+  **Meaning:** The Safe nonce is updated.<br/>
+  **Additional Note:** This number may be slightly different if other transactions have recently
+  been executed. The important thing is that it should change by 1.
```
Comment (mds1) [inline tasks/eth/2-add-superchainConfig/Validation.md:112]: nit
```suggestion
  been executed. The important thing is that it should increment by 1.
```

File: tasks/eth/2-add-superchainConfig/Validation.md [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+- **Key:** `0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e` <br/>
+  **Before:** `0x0000000000000000000000002150bc3c64cbfddbac9815ef615d6ab8671bfe43` <br/>
+  **After:** `0x000000000000000000000000a95b24af19f8907390ed15f8348a1a5e6ccbc5c6` <br/>
+  **Meaning:** The name `OVM_L1CrossDomainMessenger` is set to the address of the new
+  `L1CrossDomainMessenger`. The correctness of this slot can be validated by verifying that the **Before** address matches both of the following cast calls
+  (please consider changing out the rpc url):
+  1. what is returned by calling `AddressManager.getAddress()`:
+   ```
+   cast call 0xde1fcfb0851916ca5101820a69b13a4e276bd81f 'getAddress(string)' 'OVM_L1CrossDomainMessenger' --rpc-url https://ethereum.publicnode.com
```
Comment (mds1) [inline tasks/eth/2-add-superchainConfig/Validation.md:135]:
```suggestion
   cast call 0xde1fcfb0851916ca5101820a69b13a4e276bd81f 'getAddress(string)(address)' 'OVM_L1CrossDomainMessenger' --rpc-url https://ethereum.publicnode.com
```

File: tasks/eth/2-add-superchainConfig/Validation.md [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  **After:** `0x000000000000000000000000ababe63514ddd6277356f8cc3d6518aa8bdeb4de` <br/>
+  **Meaning:** Implementation address is set to the new [`OptimismPortal`](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0-rc.1/op-chain-ops/cmd/op-upgrade-extended-pause/main.go#L39-L42).
+
+### `0xde1fcfb0851916ca5101820a69b13a4e276bd81f` (`AddressManager`)
+
+- **Key:** `0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e` <br/>
+  **Before:** `0x0000000000000000000000002150bc3c64cbfddbac9815ef615d6ab8671bfe43` <br/>
+  **After:** `0x000000000000000000000000a95b24af19f8907390ed15f8348a1a5e6ccbc5c6` <br/>
+  **Meaning:** The name `OVM_L1CrossDomainMessenger` is set to the address of the new
+  `L1CrossDomainMessenger`. The correctness of this slot can be validated by verifying that the **Before** address matches both of the following cast calls
```
Comment (mds1) [inline tasks/eth/2-add-superchainConfig/Validation.md:131]: But how do we verify the after address?

File: tasks/eth/2-add-superchainConfig/Validation.md
Comment (maurelian) [reply tasks/eth/2-add-superchainConfig/Validation.md:1]: They are meant to indicate the order in which they are executed, not map to the upgrade numbering. 
There have been several upgrades which did not involve contracts, so they aren't reflected here.
We should clarify/improve on this system when we get a chance.

Comment (mds1) [reply tasks/eth/2-add-superchainConfig/Validation.md:1]: Got it, now tracking in https://github.com/ethereum-optimism/superchain-ops/issues/60

File: tasks/eth/2-add-superchainConfig/Validation.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+  **After:** `0x000000000000000000000000ababe63514ddd6277356f8cc3d6518aa8bdeb4de` <br/>
+  **Meaning:** Implementation address is set to the new [`OptimismPortal`](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0-rc.1/op-chain-ops/cmd/op-upgrade-extended-pause/main.go#L39-L42).
+
+### `0xde1fcfb0851916ca5101820a69b13a4e276bd81f` (`AddressManager`)
+
+- **Key:** `0x515216935740e67dfdda5cf8e248ea32b3277787818ab59153061ac875c9385e` <br/>
+  **Before:** `0x0000000000000000000000002150bc3c64cbfddbac9815ef615d6ab8671bfe43` <br/>
+  **After:** `0x000000000000000000000000a95b24af19f8907390ed15f8348a1a5e6ccbc5c6` <br/>
+  **Meaning:** The name `OVM_L1CrossDomainMessenger` is set to the address of the new
+  [`L1CrossDomainMessenger`](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0-rc.1/op-chain-ops/cmd/op-upgrade-extended-pause/main.go).
```
Comment (mds1) [inline tasks/eth/2-add-superchainConfig/Validation.md:131]:
```suggestion
  [`L1CrossDomainMessenger`](https://github.com/ethereum-optimism/optimism/blob/op-contracts/v1.2.0-rc.1/op-chain-ops/cmd/op-upgrade-extended-pause/main.go#L27-L30).
```

**Review by mds1 — APPROVED** [review]

**Pull request merged**